admob-latest/src/test/java/com/rakuten/tech/mobile/admob/RealAdmobDisplayerLatestSpec.kt

```diff
@@ -0,0 +1,111 @@
+package com.rakuten.tech.mobile.admob
+
+import android.app.Activity
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.android.gms.ads.interstitial.InterstitialAd
+import com.google.android.gms.ads.rewarded.RewardedAd
+import org.amshove.kluent.any
+import org.amshove.kluent.mock
+import org.amshove.kluent.shouldBe
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.kotlin.verify
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class RealAdmobDisplayerLatestSpec {
+
+    private lateinit var context: Activity
+    private lateinit var ad20Displayer: RealAdmobDisplayerLatest
+    private var mockOnCallback = mock<(AdStatus, String?) -> Unit>()
+    private var mockInterstitialAd: InterstitialAd = mock()
+    private var mockRewardedAd: RewardedAd = mock()
+
+    @Before
+    fun setup() {
+        ActivityScenario.launch(TestActivity::class.java).onActivity { activity ->
```

Comment:
I think you don't need to use instrumented tests for this. Instead, you could just inject the functions which you need to mock. For example in `RealAdmobDisplayerLatest`:

```
internal class RealAdmobDisplayerLatest(
    private val context: Activity,
    private val loadInterstitialAd: (Context, String, AdRequest, InterstitialAdLoadCallback) -> Unit = InterstitialAd::load,
    // etc...
) : AdmobDisplayerLatest() {
```